pkg/controller/clusterdeployment/installconfigvalidation.go

```diff
@@ -0,0 +1,50 @@
+package clusterdeployment
+
+import (
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	installertypes "github.com/openshift/installer/pkg/types"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+)
+
+const (
+	noAWSPlatformErr   = "install config did not contain an AWS platform"
+	noGCPPlatformErr   = "install config did not contain a GCP platform"
+	noAzurePlatformErr = "install config did not contain an Azure platform"
+	regionMismatchErr  = "install config region does not match cluster deployment region"
+)
+
+func ValidateInstallConfig(cd *hivev1.ClusterDeployment, installConfig []byte) error {
+
+	ic := &installertypes.InstallConfig{}
+	if err := yaml.Unmarshal(installConfig, &ic); err != nil {
+		return errors.Wrap(err, "could not unmarshal InstallConfig")
```

Comment:
Tried to do this without parsing to a type given our previous goal of limited interaction with install config as we may not have the correct version. Had a really hard time working with nested json in this manner and eventually went to the type as we had it vendored regardless. Idea here would be that the region structs are stable and should not change, and it doesn't matter if the install config has features we don't have vendored yet, we just want to check some core things.